src/commands/infra/mocks/definition.json

```diff
@@ -0,0 +1,42 @@
+{
+  "name": "discovery-service",
+  "source": "https://github.com/Microsoft/bedrock.git",
+  "template": "cluster/environments/azure-single-keyvault",
+  "version": "0.0",
+  "backend": {
+    "storage_account_name": "storage-account-name",
+    "access_key": "storage-account-access-key",
+    "container_name": "storage-account-container",
+    "key": "tfstate-key"
+  },
+  "variables": {
+    "acr_enabled": "true",
+    "address_space": "<insert value>",
+    "agent_vm_count": "<insert value>",
+    "agent_vm_size": "<insert value>",
+    "cluster_name": "<insert value>",
+    "dns_prefix": "<insert value>",
+    "flux_recreate": "<insert value>",
+    "kubeconfig_recreate": "<insert value>",
+    "gc_enabled": "true",
+    "gitops_poll_interval": "5m",
+    "gitops_ssh_url": "<insert value>",
+    "gitops_url_branch": "master",
+    "gitops_ssh_key": "<insert value>",
+    "gitops_path": "<insert value>",
+    "keyvault_name": "<insert value>",
+    "keyvault_resource_group": "<insert value>",
+    "resource_group_name": "<insert value>",
+    "ssh_public_key": "<insert value>",
+    "service_principal_id": "<insert value>",
+    "service_principal_secret": "<insert value>",
+    "subnet_prefixes": "<insert value>",
+    "vnet_name": "<insert value>",
+    "subnet_name": "<insert value>",
+    "network_plugin": "azure",
+    "network_policy": "azure",
+    "oms_agent_enabled": "false",
+    "enable_acr": "false",
+    "acr_name": "<insert value>"
+  }
+}
```

Comment:
Committing a "mock" definition.json file to master branch for the sake of unit testing might not be ideal, especially if definition.json files could vary. Something we have done in `spk infra scaffold` is create a JSON object as part of the unit test (instead of relying on a file), and testing on components from the JSON object. Thoughts?

Reply:
Refer to this: https://github.com/CatalystCode/spk/blob/baf4adb74317295c1e057b965ba49df3834a82a8/src/commands/infra/scaffold.test.ts#L18

Reply:
The unit test checks the presence of a definition.json in the provided project folder. From there it checks the source url for a repository. I can see the definition.json becoming more complex for future unit tests we want to use.